### PR TITLE
create registry credentials when gcp-auth addon is enabled

### LINKED ISSUE #9002: gcp-auth plugin should support image pulls from gcr.io
As reported in #Slack and seen within two friction logs.

### PR DESCRIPTION
**Pull request by sharifelgamal**
Fixes #9002 

This PR is dependent on https://github.com/GoogleContainerTools/gcp-auth-webhook/pull/11 and a new release of the gcp-auth-webhook image.

### CODE REVIEW

**Review by ilya-zuyev — APPROVED** [review]
lgtm

File: cmd/minikube/cmd/config/profile_list.go [hunk 1]
```diff
@@ -148,7 +148,7 @@ func profileStatus(p *config.Profile, api libmachine.API) string {
 
 func renderProfilesTable(ps [][]string) {
 	table := tablewriter.NewWriter(os.Stdout)
-	table.SetHeader([]string{"Profile", "VM Driver", "Runtime", "IP", "Port", "Version", "Status", "Nodes"})
+	table.SetHeader([]string{"Profile", "Driver", "Runtime", "IP", "Port", "Version", "Status", "Nodes"})
```
Comment (medyagh) [inline cmd/minikube/cmd/config/profile_list.go:151]: unrelated changes in separte PRs

Comment (sharifelgamal) [reply cmd/minikube/cmd/config/profile_list.go]: done

**Review by medyagh — CHANGES_REQUESTED** [review]
lets add an integration test for this, when only running the test in GCP to try to pull an image from a private GCR that k8s-minikube project has access to .
and skip the imge if not running in GCP

File: test/integration/addons_test.go [hunk 2]
```diff
@@ -46,17 +46,20 @@ func TestAddons(t *testing.T) {
...
+		defer os.Unsetenv("GOOGLE_APPLICATION_CREDENTIALS")
+		if err != nil {
+			t.Fatalf("Failed setting GOOGLE_APPLICATION_CREDENTIALS env var: %v", err)
+		}
 
-	err = os.Setenv("GOOGLE_CLOUD_PROJECT", "this_is_fake")
-	defer os.Unsetenv("GOOGLE_CLOUD_PROJECT")
-	if err != nil {
-		t.Fatalf("Failed setting GOOGLE_CLOUD_PROJECT env var: %v", err)
+		err = os.Setenv("GOOGLE_CLOUD_PROJECT", "this_is_fake")
```
Comment (medyagh) [inline test/integration/addons_test.go:58]: isn't this overridng the preivous os.SetEnv ? did u mean to be in a different block ?

Comment (sharifelgamal) [reply test/integration/addons_test.go:58]: i think the diff is a little confusing, but this is doing what it always has been, just inside of an if statement

**Review by medyagh — APPROVED** [review]
